### DISCUSSION TITLE
Enable / Disable face recognition via occ command

### DISCUSSION THREAD

**Discussion by Schema-F** [nextcloud/recognize#943] [Q&A] [answered]
Is it possible to enable / disable face recognition and / or object detection with a CLI command?
I want to run the detection only through specific hours a day with crontab.

**Comment by marcelklehr** [comment, marked as answer]
> I want to run the detection only through specific hours a day with crontab.

You can achieve this by setting a maintenance window (see https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/background_jobs_configuration.html#parameters )

Alternatively you can use

* `occ config:app:set recognize faces.enabled --value "true"`
* `occ config:app:set recognize faces.enabled --value "false"`
* `occ config:app:set recognize imagenet.enabled --value "true"`
* `occ config:app:set recognize imagenet.enabled --value "false"`

Note however, that these config keys may change at any time without warning.

**Comment by khlschrnk** [comment]
I am trying to find a way to let recognize enabled all the time but to let actual face recognition run only during time my photovoltaic system has enough energy to cover this part of the added electricity usage.

I'd like to have recognize enabled all the time as only this allows me to see the "people" tab in memories.

I am looking for a command to e.g. set the number of to be processed files to a higher number once my homeassistant shows surplus energy and set it a very low number once no engery is produced. This does not affect useability of the server in a significant ways as I have restricted the number of cores to use.

The maintenance window does not seem to be suitable as then other processes are run as well and I would expect a slowed experience.

Any ideas?

**Reply by marcelklehr** [reply]
You can set the batch sizes similarly as the above:
```
    occ config:app:set recognize faces.batchSize --value "1"
    occ config:app:set recognize imagenet.batchSize --value "1000"
    occ config:app:set recognize landmarks.batchSize --value "500"
```

**Reply by khlschrnk** [reply]
Thank you, that helps!
